### PR TITLE
DOC: Example for integer overflow

### PR DESCRIPTION
**Pull request by vitrioil**
As suggested in #21326, adding an example for getting negative square results for integers.

### CODE REVIEW

File: doc/source/user/basics.types.rst [hunk 1]
```diff
@@ -237,6 +237,11 @@ range of possible values.
     >>> np.power(100, 100, dtype=np.float64)
     1e+200
 
+    >>> np.power(53782400000 - 61917364224, 2) # Integer arithmetic is modular
+    -7609333369078284288
+    >>> np.power(53782400000.0 - 61917364224.0, 2) # Prefer floating point for computation
+    6.617764292575992e+19
+
```
[the remark this answers is not in the document]
Comment (vitrioil) [reply doc/source/user/basics.types.rst:244]: I will combine this with the previous block.

What do you think of an example like this:
```python3
>>> np.power(int(2**32)+1, 2)
8589934593
>>> np.power(int(2**32)-1, 2)
-8589934591
>>> np.power(float(2**32)+1, 2)
1.8446744082299486e+19
```

**Comment by rossbar** [comment]
This one's been stale for a while, and given the redundancy with the two preceding examples I'm inclined to close it.

**Comment by rossbar** [comment]
This was discussed at the documentation meeting and the general consensus was that the overflow cases were sufficiently covered in the existing doc immediately above this. Thanks for the proposal @vitrioil !

**Comment by mdhaber** [comment]
@rossbar does that mean that gh-21326 can be closed, or is there room with the existing overflow examples for one more showing that the result of a positive integer to an integer power can be negative?

**Pull request closed without merging**